src/frontend/sharedComponents/ProjectInviteBottomSheet/index.tsx

```diff
@@ -1,145 +1,230 @@
-          accept={accept}
-        />
-      </BottomSheetModal>
+      <BottomSheetModalProvider>
+        <BottomSheetModal
+          ref={inviteBottomSheet.sheetRef}
+          isOpen={inviteBottomSheet.isOpen}>
+          {displayedInvite ? (
+            <InviteBottomSheetContent
+              sessionInvite={displayedInvite}
+              startConfirmationFlow={() => {
+                inviteBottomSheet.closeSheet();
+                leaveProjectSheet.openSheet();
+              }}
+              onAccept={() => {
+                setDisplayedInviteId(undefined);
+                inviteBottomSheet.closeSheet();
+              }}
+              onDismiss={seeNextInviteOrClose}
+              onReject={() => {
+                lastRejectedInviteIdRef.current =
+                  displayedInvite.invite.inviteId;
+                seeNextInviteOrClose();
+              }}
+            />
+          ) : (
+            // Using null can sometimes cause a rendering issue with bottom sheet modal
+            <View style={{height: 100}} />
+          )}
+        </BottomSheetModal>
+      </BottomSheetModalProvider>
+
+      <BottomSheetModalProvider>
+        <BottomSheetModal
+          fullScreen
+          ref={leaveProjectSheet.sheetRef}
+          isOpen={leaveProjectSheet.isOpen}>
+          {displayedInvite ? (
+            <LeaveProjectModalContent
+              onCancel={() => {
+                leaveProjectSheet.closeSheet();
+                inviteBottomSheet.openSheet();
+              }}
+              onSuccess={() => {
+                leaveProjectSheet.closeSheet();
+                inviteBottomSheet.openSheet();
+              }}
+              inviteId={displayedInvite.invite.inviteId}
+              projectName={displayedInvite.invite.projectName}
+            />
+          ) : (
+            // Using null can sometimes cause a rendering issue with bottom sheet modal
+            <View style={{height: 100}} />
+          )}
+        </BottomSheetModal>
+      </BottomSheetModalProvider>
     </>
   );
 };
+
+function useSessionInvites() {
+  const pendingInvitesQuery = usePendingInvites();
+  const removedInvites = useRemovedInvite();
+
+  const pending = pendingInvitesQuery.data
+    .filter(invite => removedInvites.data?.invite.inviteId !== invite.inviteId)
+    .map(invite => ({
+      status: 'pending' as const,
+      invite,
+    }));
+
+  return {
+    pending,
+    removed: removedInvites.data,
+    clearRemoved: removedInvites.clear,
+  };
+}
+
+function useRemovedInvite() {
+  const mapeoApi = useApi();
+
+  const [removedInvite, setRemovedInvite] = React.useState<{
+    invite: MapBuffers<InviteInternal>;
+    reason: InviteRemovalReason;
+  } | null>(null);
+
+  React.useEffect(() => {
+    function onInviteRemoved(
+      invite: MapBuffers<InviteInternal>,
+      reason: InviteRemovalReason,
+    ) {
+      setRemovedInvite(prev => {
+        if (prev) return prev;
+        return {invite, reason};
+      });
```

Comment:
we only keep a single removed invite in state at a time. clearing it becomes the responsibility of an (ideally explicit) client-side interaction. this allows us to still properly reference an invite after it's been updated.